### ISSUE TITLE
Is there a way to pass the regexp that was available in the old goji?

### ISSUE THREAD

**Issue by catatsuy** [goji/goji#67]
I have defined the following endpoint with https://github.com/zenazn/goji

```
goji.Get(regexp.MustCompile(`^/@(?P<accountName>[a-zA-Z]+)$`), getAccountName)
```

Can I get the same behavior with the new goji?

**Comment by zenazn** [comment]
Yes! But it's not included with the base library any more. Instead, the new version of Goji makes it easy to support custom `Pattern`s.

In order to get this functionality, you'll have to write your own implementation of https://pkg.go.dev/goji.io?utm_source=godoc#Pattern. A basic implementation would:
- Get the path from the request context using https://pkg.go.dev/goji.io@v2.0.2+incompatible/pattern#Path
- Run that path through your regexp of choice
- Make a custom `Context` implementation that implements the following `Value` protocol:
  - `Variable` support: https://pkg.go.dev/goji.io@v2.0.2+incompatible/pattern#Variable
  - `AllVariables ` support: https://pkg.go.dev/goji.io@v2.0.2+incompatible/pattern#Variable
  - (here's `pat`'s implementation of that protocol: https://github.com/goji/goji/blob/490b001d03d16a116d7c44d95d216c69a1aae798/pat/match.go)

Let me know if you end up writing this, and if you get stuck along the way I'd be happy to help out!

**Comment by catatsuy** [comment]
@zenazn Thank you! I tried it and I have implemented the following code. Is this the right way to use it?

```go
type RegexpPattern struct {
	regexp *regexp.Regexp
}

func Regexp(reg *regexp.Regexp) *RegexpPattern {
	return &RegexpPattern{regexp: reg}
}

func (reg *RegexpPattern) Match(r *http.Request) *http.Request {
	ctx := r.Context()
	uPath := pattern.Path(ctx)
	if reg.regexp.MatchString(uPath) {
		values := reg.regexp.FindStringSubmatch(uPath)
		keys := reg.regexp.SubexpNames()

		for i := 1; i < len(keys); i++ {
			ctx = context.WithValue(ctx, pattern.Variable(keys[i]), values[i])
		}

		return r.WithContext(ctx)
	}

	return nil
}
```
